File: Practica_4/civitasJava/src/civitas/SorpresaEspeculador.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;
 
/**
 * 
 * @author mariolopezgonzalez
 */
public class SorpresaEspeculador extends Sorpresa{
    
    SorpresaEspeculador(int fianza, String texto){
       super(texto, fianza, null);
    }
    
    @Override
    void aplicarAJugador(int actual, ArrayList<Jugador> todos) {
        if (jugadorCorrecto(actual, todos)) {
            informe(actual, todos);
            Jugador jugador = todos.get(actual);
            JugadorEspeculador jugadorEspeculador = new JugadorEspeculador(jugador, super.getValor());
            todos.set(actual, jugadorEspeculador);
        }
    }

    @Override
    public String toString() {
        return "SorpresaEspeculador{ "+ super.getTexto() + ", fianza=" + super.getValor() + '}';
    }
}

File: EXAMEN/EXAMEN 2/civitasRuby/lib/sorpresa_especulador.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
 
module Civitas
  class Sorpresa_especulador < Sorpresa
    def initialize (fianza, texto)
      super(texto, fianza)
    end

    # Método aplicarAJugador (sorpresa especulador)
    # Parámetros:
    #   actual: indice del jugador actual
    #   todos: array que contiene los jugadores
    # Efecto:
    #   Envia al jugador actual a la nueva casilla
    def aplicar_a_jugador(actual, todos)
      if jugador_correcto(actual, todos)
        informe(actual, todos)
        todos[actual] = Jugador_especulador.new(todos[actual], valor)
      end
    end

    def to_s
      return "Sorpresa Especulador ->  { Texto: #{texto}, Fianza #{valor} }"
    end

    public_class_method :new

  end
end
